mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -291,6 +295,61 @@ impl NvmeIoChannelInner<'_> {
         self.qpair = Some(qpair);
         0
     }
+
+    /// Get I/O statistics for channel.
+    #[inline]
+    pub fn get_io_stats_controller(&mut self) -> &mut IoStatsController {
+        &mut self.io_stats_controller
+    }
+}
+pub struct IoStatsController {
+    io_stats: BlockDeviceIoStats,
+    block_size: u64,
+}
+
+/// Top-level wrapper around device I/O statistics.
+impl IoStatsController {
+    fn new(block_size: u64) -> Self {
+        Self {
+            io_stats: BlockDeviceIoStats::default(),
+            block_size,
+        }
+    }
+
+    #[inline]
+    /// Account amount of blocks and I/O operations.
+    pub fn account_block_io(
+        &mut self,
+        op: IoType,
+        num_ops: u64,
+        num_blocks: u64,
+    ) {
+        let num_bytes = num_blocks * self.block_size;
```

Comment:
Not sure how much raw the stats should be. But I personally would prefer as raw numbers as possible and I can do the remaining math as a consumer of the stats. Also the code in IO path that is called often, could be simpler then. So if we recorded just number of reads and writes we could save a couple of instructions and a couple of bytes of memory. Assuming the consumer knows the block size of the device, she can calculate the rest easily. Just an idea and it's really a micro optimisation.

Reply:
Actually, libSPDK does the same as we do, they calculate the real stats in place:

bdev_io->internal.ch->stat.bytes_read += bdev_io->u.bdev.num_blocks * bdev_io->bdev->blocklen;

Looks like they don't care much about the speed. :-)

But not a problem, I can move multiplication step to 'get_stats()' and return updated stats when explicitly requested. I would prefer to keep translation as close to the stats itself as possible and not to put any burden to the caller.

Reply:
It appears that all callers pass 1 to `num_ops` so perhaps that can be dropped.